h/security/permits/hybrid.py

```diff
@@ -0,0 +1,144 @@
+from enum import Enum
+
+from h.security.permits.predicates import *
+
+
+class PermissionEnum(Enum):
+    def __init__(self, _value, predicates):
+        self.predicates = predicates
+
+
+class Group(PermissionEnum):
+    ADMIN = (
+        "group:admin",
+        [
+            [group_authority_matches_authenticated_client],
+            [group_created_by_user],
+            # Those with the admin or staff Role should be able to admin/edit any
+            # group
+            [user_is_staff],
+            [user_is_admin],
+        ],
+    )
```

Comment:
This is an example of adding the predicates directly to the permissions Enums instead of maintaining an extra list.

It makes this longer but it's shorter than the sum of the two, and I like the localism of defining a permission _and_ how it's granted in the same place. If we took the time to add a doc string for each of these it would be a very good resource.